errors/error.go

```diff
@@ -69,8 +69,11 @@ const separator = ": "
 //		The kind of error (eg.: HCLSyntax, TerramateSchema, etc).
 //	hcl.Range
 //		The file range where the error originated.
-//		errors.StackMeta
+//	errors.StackMeta
 //		The stack that originated the error.
+//  *List
+//		The underlying error list. By wrapping an error list, you also wrap all
```

Comment:
Maybe add some of the info from the internal comment to here, the docs here are a little confusing on what exactly will be the behavior, like what does "you also wrap all of its error items" means ? That doesn't make it clear to me that all args passed to E() will be propagated to each of the errors on the list.